scripts/README.md

```diff
@@ -2,4 +2,71 @@
 
 All of the staged repositories live in the top level `staging` directory.
 
-Note: the process for porting upstream commits downstream is outlined in the [OLM downstreaming guide](https://docs.google.com/document/d/139yXeOqAJbV1ndC7Q4NbaOtzbSdNpcuJan0iemORd3g/edit#).
+The downstreaming process is complex and helper scripts have been written
+to facilitate downstreaming.
+
+## Assumptions
+
+The helper scripts assume that the upstream remote repos are configured
+as follows:
+```
+git remote add api https://github.com/operator-framework/api
+git remote add operator-registry https://github.com/operator-framework/operator-registry
+git remote add operator-lifecycle-manager https://github.com/operator-framework/operator-lifecycle-manager
+```
+The [sync.sh](sync.sh) script will automatically create these
+remote repositories.
+
+## Bulk Sync
+
+To sync all current changes from upstream, simply run the sync script:
+```sh
+scripts/sync.sh
+```
+
+This script may pause at certain points to ask the user to examine
+command failures or possible regressions. Please open another terminal
+and review the state of the workspace before continuing the script.
+
+When the script completes, it will have created a branch whose name is
+the current date (formatted: `sync-YYYY-MM-DD`). Create a PR from this
+sync branch.
+
+If the script fails, please refer to [[1](https://spaces.redhat.com/display/OOLM/Downstream+to+operator-framework-olm)]
+for continuation proceedures.
+
+## Targeted Sync
+
+To sync a subset of commits from the upstream repositories (i.e. critical
+bugfix), create a new working sync branch. Then create a `sync.cherrypick`
+file in the repositry root directory with the repos and commit SHAs.
+
+The format of the cherrypick file is:
+```
+<order> <repo> <commit-SHA>
+```
+
+For example:
+```
+1 api 0123456789abcdef0123456789abcdef01234567
+2 operator-lifecycle-manager 123456789abcdef0123456789abcdef012345678
+3 operator-lifecycle-manager 23456789abcdef0123456789abcdef0123456789
+```
+Do _not_ commit the cherrypick file, it is a temporary working file that
+is ignored by `git`.
+
+Then run the following:
+```sh
+scripts/sync_pop_candidate.sh -a sync
+```
+The commits in the `sync.cherrypick` file will be applied in the specified
+order.
+
+Even if you only have a single commit, this procedure will follow the same
+process that `sync.sh` does, to ensure no steps are missed.
+
```

Comment:
Doesn't sync_pop_candidate.sh also make manifests, vendor, etc. in ways that sync.sh does not?  
(And also, sync_pop_candidate.sh needs to make verify-manifests in order to satisfy CI.)

Reply:
I'm going to document the need to do a `make verify` but fixing the downstreaming scripts to do that is a bit more involved, so I'd rather save that for another PR.

Reply:
My point was really more about the statement about 'will follow the same process' when there is a behavior gap between the scripts. 
for e.g.:
- [make manifests](https://github.com/openshift/operator-framework-olm/blob/9a977fa7b6c40ad237abc4645dc41e58cafbb944/scripts/sync_pop_candidate.sh#L109)
- [updating commit format to track commit sources](https://github.com/openshift/operator-framework-olm/blob/9a977fa7b6c40ad237abc4645dc41e58cafbb944/scripts/sync_pop_candidate.sh#L108)

Reply:
But _it is_ the same process. The sync.sh script:

1. The upstream repositories are added as remotes to the monorepo.
2. The upstream repositories are fetched.
3. A merge branch with the current date (sync-YYYY-MM-DD) is created to do the bulk sync.
4. Create a list of candidate commits from upstream to downstream via `sync_get_candidates.sh`
5. Apply each commit via `sync_pop_candidate.sh`

The `sync.sh` script literally calls `sync_pop_candidate.sh` to do all the work for a list of commits (generated by `sync_get_candidates.sh`.

The problem is that there's additional verification steps via `make verify`/`make verify-manifests`, that need to occur.

Reply:
Argh!  You're completely correct.  I missed the call to sync_pop_candidates.sh. ☹️ 
Thanks for the sanity.

Reply:
IMHO we should solve the verification "additional steps" problem in changes to sync_pop_candidates.sh in another PR.

Reply:
Agreed.